spec/clearance/back_door_spec.rb

```diff
@@ -46,6 +46,18 @@
     end
   end
 
+  it "strips 'as' from the params" do
+    user_id = "123"
+    user = double("user")
+    allow(User).to receive(:find).with(user_id).and_return(user)
+    env = env_for_user_id(user_id)
+    back_door = Clearance::BackDoor.new(mock_app)
+
+    back_door.call(env)
+
+    expect(env["QUERY_STRING"]).to be_empty
```

Comment:
Will you also need to test for form data in POST requests? I don't think they'll end up in the query string.

Reply:
The middleware always looks in the query string, even before this PR. Unless I'm misunderstanding what Rack puts in `env["QUERY_STRING"]` the middleware wouldn't find an `as` parameter in a POST body.

Reply:
From the [Rack SPEC], it looks like it won't put form-encoded data in `QUERY_STRING`, but looking at an example of using `as` in an application, I see that we're using it in the URL builder methods and not when building POST bodies, so this seems like a non-issue.

[Rack SPEC](https://www.rubydoc.info/github/rack/rack/file/SPEC)